File: src/main/java/com/SGTPI/SystemProject/dto/AppointmentRequestDto.java

package com.SGTPI.SystemProject.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.Valid;
import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotBlank;
import java.time.LocalDate;
import java.time.LocalTime;


public record AppointmentRequestDto(
        @JsonProperty("duration")
        int duration,
        @JsonProperty("fecha")
        @JsonFormat(pattern = "yyyy/MM/dd")
        @Future(message = "La fecha debe ser futura")        
        LocalDate  fecha,
        @JsonProperty("hora")
        @JsonFormat(pattern = "HH:mm:ss")
        LocalTime  hora,
        @JsonProperty("patient")
        @Valid
        @NotBlank(message = "El paciente es obligatorio")
        PatientDto patient
        ) {
    
}

File: FrontEnd/Angular/src/app/interfaces/AppointmentRequestDto.ts
import { AppointmentPatientDto } from './AppointmentPatientDto';

export interface AppointmentRequestDto {
  fecha: string; // Formato YYYY/MM/DD
  hora: string; // Formato HH:MM:SS
  duration: number;
  patient: AppointmentPatientDto; // El paciente debe ser un objeto AppointmentPatientDto
  state: string;
  sessionNotes?: string;
}
